app/jobs/characterize_job.rb

```diff
@@ -2,9 +2,10 @@ class CharacterizeJob < ActiveJob::Base
   queue_as CurationConcerns.config.ingest_queue_name
 
   # @param [FileSet] file_set
-  # @param [String] filename a local path for the file to characterize. By using this, we don't have to pull a copy out of fedora.
+  # @param [String] filename a local path for the file to characterize so we don't have to pull a copy out of fedora.
   def perform(file_set, filename)
-    Hydra::Works::CharacterizationService.run(file_set, filename)
+    return unless file_set.characterization_proxy.content
```

Comment:
Do we want to be asking for the content? Won't that read the entire binary? Would it make sense to have a `file_set.characterization_proxy?` method or some such check that can prevent the more intimate knowledge of the workings of a characterization_proxy?

In reading the code, I see a guard of `file_set.characterization_proxy.content` followed by usage of `file_set.characterization_proxy`. This is a bit of a smell in that the guard has more intimate knowledge of the model than the service that is being called on the next line.